### PR TITLE
Introduce "HTTPRequest" class

### PR DESCRIPTION
**Pull request by MrPetovan**
Redux of #8364
Addon side: https://github.com/friendica/friendica-addons/pull/1006

This branch is currently running on my test node.

### CODE REVIEW

File: mod/dfrn_poll.php [hunk 1]
```diff
@@ -499,20 +498,20 @@ function dfrn_poll_content(App $a)
...
 			// URL reply
 			if ($dfrn_version < 2.2) {
-				$s = Network::fetchUrl($r[0]['poll']
-					. '?dfrn_id=' . $encrypted_id
-					. '&type=profile-check'
-					. '&dfrn_version=' . DFRN_PROTOCOL_VERSION
-					. '&challenge=' . $challenge
-					. '&sec=' . $sec
+				$s = DI::httpRequest()->fetch($r[0]['poll']
+				                              . '?dfrn_id=' . $encrypted_id
```
Comment (annando) [inline mod/dfrn_poll.php:502]: Just some quick note: Is the indention correct?

Comment (MrPetovan) [reply mod/dfrn_poll.php:502]: Apparently so, to align with `$r[0]['poll']`. I was in the middle of a massively conflicted rebase with 18 steps, so I didn't touch the parts that weren't conflicting!

**Comment by MrPetovan** [comment]
To my disappointment, this PR will not unblock #8845. We will need another massive refactoring to change all the module classes to use PSR-7 style messages in input and output, and this can only happen when #4090 is done.

**Comment by nupplaphil** [comment]
@MrPetovan That's where I got stuck too :-/ .. PSR-7 compliant means returning this type of interfaces. Either with a library (I started with https://github.com/guzzle/psr7 ) or new classes

**Comment by MrPetovan** [comment]
Aren't both parts (Use Guzzle and make modules use PSR-7 messages) entirely separate, though? Thanks for the insight.

**Comment by MrPetovan** [comment]
Found a couple of stragglers, this PR is now ready to be merged.

**Comment by MrPetovan** [comment]
Still no errors.

**Comment by annando** [comment]
I will have a loot at it this evening (MEST), yesterday I was too tired (I hunted for the comet the two days before deep in the night) and went to bed early.

**Comment by MrPetovan** [comment]
It's okay. It is somewhat complicating my workflow because I have this branch running on my test node but I can manage.

**Pull request merged**